src/charm.py

```diff
@@ -81,9 +80,9 @@ def change_config(self, _: ops.HookEvent) -> None:
             self.pebble_service.change_config(container)
         except (
             CharmConfigInvalidError,
```

Comment:
Perhaps these could be grouped by a parent exception? Exception granularity is useful if we do something different for each one, a long list like this becomes more difficult to maintain and reason about (e.g., which one is and isn't covered)

Reply:
Your comment gave me this idea and I think is kind of related to something that @gregory-schiano suggested. I created an actions module (not with observers, only a module to extract the interactions from the charm) so this made it possible to create specific exceptions like actions.ResetInstanceError for example. Please take a look and tell me what you think. I'll close this comment since is outdated compared to the change that I made.